python/grass/gunittest/loader.py

```diff
@@ -235,4 +235,18 @@
 
 
 if __name__ == "__main__":
-    GrassTestLoader().discover()
+    for expression in [r".*\.py$", r".*\.sh$"]:
+        modules = discover_modules(
+            start_dir=".",
+            grass_location="all",
+            file_regexp=expression,
+            skip_dirs=GrassTestLoader.skip_dirs,
+            testsuite_dir=GrassTestLoader.testsuite_dir,
+            all_locations_value=GrassTestLoader.all_tests_value,
+            universal_location_value=GrassTestLoader.universal_tests_value,
+            import_modules=False,
+            exclude=None,
+        )
+        print("Expression:", expression)
+        print(len(modules))
+        print(list(module.file_path for module in modules))
```

Comment:
Wait, that's weird, it seems like https://github.com/astral-sh/ruff/issues/10778, but the conclusion was that the rules don't conflict.

Reply:
The error log says `[error]python/grass/gunittest/loader.py:252:15: C400 Unnecessary generator (rewrite as a `list` comprehension)` so yes, [C400 unnecessary-generator-list](https://docs.astral.sh/ruff/rules/unnecessary-generator-list/). I don't see a problem replacing `list()` with `[]`.

Reply:
Usually, when you run ruff locally with (let's say with --preview flag), it'll apply the fix, and then a fix for the C416 that it might generate afterwards. Both would be applied at the same time when using --fix (a fix leading to another fixable error)